htmresearch/support/nlp_model_test_helpers.py

```diff
@@ -0,0 +1,242 @@
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+helpStr = """
+Methods and data for running NLP model API tests. The intent here is
+to ensure and changes to the models does not decrease their classification
+accuracies (see nlpModelAccuracies below). Three tests are supported:
+
+  hello classification: Very simple, hello world classification test. There are
+    two categories that can be discriminated using bag of words. The training
+    set is 8 docs, and the test set is an additional 2 (i.e. 10 total) -- first
+    is an incorrectly labeled version of a training sample, second is
+    semantically similar to one of the training samples.
+  simple queries: Qualitative test where we query a trained model to see which
+    data samples are the most and least similar.
+  simple labels: Less simple classification test. The dataset used here must be
+    specified in the command line args.
+
+"""
+import argparse
+import numpy
+
+from prettytable import PrettyTable
+from textwrap import TextWrapper
+from tqdm import tqdm
+
+from htmresearch.frameworks.nlp.classification_model import ClassificationModel
+from htmresearch.frameworks.nlp.model_factory import (
+  createModel, getNetworkConfig)
+
+
+
+# There should be one "htm" model for each htm config entry.
+nlpModelTypes = [
+  "docfp",
+  "cioword",
+  "htm",
+  "htm",
+  "htm",
+  "keywords"]
+
+# Network models use 4k retina.
+htmConfigs = [
+  ("HTM_sensor_knn", "../data/network_configs/sensor_knn.json"),
+  ("HTM_sensor_simple_tp_knn", "../data/network_configs/sensor_simple_tp_knn.json"),
+  ("HTM_sensor_tm_knn", "../data/network_configs/sensor_tm_knn.json"),
+]
+
+# Some values of k we know work well.
+kValues = { "keywords": 21, "docfp": 1}
+
+nlpModelAccuracies = {
+  "hello_classification": {
+    "docfp": 90.0,
+    "cioword": 90.0,
+    "HTM_sensor_knn": 80.0,
+    "HTM_sensor_simple_tp_knn": 90.0,
+    "HTM_sensor_tm_knn": 90.0,
+    "keywords": 80.0,
+  },
+  "simple_queries": {
+    "docfp": "good but not great",
+    "cioword": "passable",
+    "HTM_sensor_knn": "passable",
+    "HTM_sensor_simple_tp_knn": "idk, no results yet!",
+    "HTM_sensor_tm_knn": "idk, no results yet!",
+    "keywords": "passable",
+  },
+  "simple_labels": {
+    "docfp": 100.0,
+    "cioword": 100.0,
+    "HTM_sensor_knn": 66.2,
+    "HTM_sensor_simple_tp_knn": 99.7,
+    "HTM_sensor_tm_knn": 0.0,
+    "keywords": 16.9,
+  },
+}
+
+wrapper = TextWrapper(width=80)
+
+
+
+def executeModelLifecycle(args, trainingData, labelRefs):
```

Comment:
What is `args`?  I see that it is passed along to `instantiateModel()`, but there's no description of its structure or expected values.